src/core/Table/TablePaginator.tsx

```diff
@@ -0,0 +1,335 @@
+import { ButtonGroup } from '../ButtonGroup';
+import { Button } from '../Buttons/Button';
+import { useOverflow } from '../utils/hooks/useOverflow';
+import { ProgressRadial } from '../ProgressIndicators';
+import { TableProps } from './Table';
+import { DropdownButton } from '../Buttons';
+import { MenuItem } from '../Menu';
+import { CommonProps, getBoundedValue, useTheme } from '../utils';
+
+const defaultLocalization = {
+  rowsPerPageLabel: (size: number) => `${size} per page`,
+  currentRowsInfoLabel: (
+    startIndex: number,
+    endIndex: number,
+    totalRows: number,
+    isLoading: boolean,
+  ) =>
+    isLoading
+      ? `${startIndex}-${endIndex}…`
+      : `${startIndex}-${endIndex} of ${totalRows}`,
+};
+
+export type TablePaginatorProps = {
+  /**
+   * The zero-based index of the current page.
+   */
+  currentPage: number;
+  /**
+   * Total number of rows.
+   */
+  totalRowsCount: number;
+  /**
+   * Number of rows per page.
+   */
+  pageSize: number;
+  /**
+   * Callback when page is changed.
+   */
+  onPageChange: (page: number) => void;
+  /**
+   * Callback when page size is changed.
+   */
+  onPageSizeChange?: (size: number) => void;
+  /**
+   * Modify the density of the pagination (adjusts the height).
+   * @default 'default'
+   */
+  density?: TableProps['density'];
+  /**
+   * Flag whether data is still loading and total rows count is not known.
+   * @default false
+   */
+  isLoading?: boolean;
+  /**
+   * Control whether focusing tabs (using arrow keys) should automatically select them.
+   * Use 'manual' if tab panel content is not preloaded.
+   * @default 'auto'
+   */
+  focusActivationMode?: 'auto' | 'manual';
+  /**
+   * Array of possible page size options. When provided then shows current rows info and page size selection.
+   */
+  rowsPerPage?: number[];
+  /**
+   * Object of functions to localize pagination labels.
+   */
+  localization?: {
+    /**
+     * Function that returns page size selection label.
+     * @default (size: number) => `${size} per page`
+     */
+    rowsPerPageLabel?: (size: number) => string;
+    /**
+     * Function that returns currently visible rows info.
+     * @default
+     *  (
+     *  startIndex: number,
+     *  endIndex: number,
+     *  totalRows: number,
+     *  isLoading: boolean,
+     *  ) =>
+     *    isLoading
+     *      ? `${startIndex}-${endIndex}…`
+     *      : `${startIndex}-${endIndex} of ${totalRows}`;
+     */
+    currentRowsInfoLabel?: (
+      startIndex: number,
+      endIndex: number,
+      totalRows: number,
+      isLoading: boolean,
+    ) => string;
+  };
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * Table paginator component. Recommended to pass to the `Table` as `paginatorRenderer` prop.
+ * @example
+ * <Table
+ *   // ...
+ *   paginatorRenderer={(props) => <TablePaginator {...props} />}
+ * />
+ */
+export const TablePaginator = (props: TablePaginatorProps) => {
```

Comment:
We discussed that this approach could help us avoid the need for adding `nativePaginatorProps` or anything similar. Because every time we go with options approach, we end up having a lot of native props.